### PR TITLE
Sony PRSTUX support

### PR DESCRIPTION
**Pull request by v01d**
This completes the support for PRSTUX.
I think the device interface is relatively simple so I have not much to say about it.
In any case, please note that this include a change already discussed in #4192 
I actually had to introduce it since otherwise koreader would completely block if it could not reach the internet.

BTW, I'm not sure what/if I need to add anything for supporting CI in this repo, as I don't fully understand the config. I would like to have PRSTUX image created for nightlies/releases as it is already the case for the rest of the devices.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -360,6 +361,32 @@ androidupdate: all
 	cp $(ANDROID_LAUNCHER_DIR)/bin/NativeActivity-debug.apk \
 		koreader-android-$(MACHINE)-$(VERSION).apk
 
+sony_prstuxupdate: all
+	# ensure that the binaries were built for ARM
+	file $(INSTALL_DIR)/koreader/luajit | grep ARM || exit 1
+	# remove old package if any	
+	rm -f koreader-sony-prstux-$(MACHINE)-$(VERSION).zip
+	# Kobo launching scripts
```
Comment (NiLuJe) [inline Makefile]: Not Kobo, obviously ;).

File: Makefile [hunk 2]
```diff
@@ -360,6 +361,32 @@ androidupdate: all
...
+	                "koreader/resources/icons/src/*" "koreader/spec/*" \
+	                $(ZIP_EXCLUDE)
+	# generate update package index file
+	zipinfo -1 koreader-sony-prstux-$(MACHINE)-$(VERSION).zip > \
+	        $(INSTALL_DIR)/koreader/ota/package.index
+	echo "koreader/ota/package.index" >> $(INSTALL_DIR)/koreader/ota/package.index
+	# update index file in zip package
+	cd $(INSTALL_DIR) && zip -u ../koreader-sony-prstux-$(MACHINE)-$(VERSION).zip \
+	        koreader/ota/package.index
+	# make gzip koboupdate for zsync OTA update
```
Comment (NiLuJe) [inline Makefile]: Same ;)

File: .gitignore [hunk 3]
```diff
@@ -51,3 +51,4 @@ koreader-emulator-x86_64-pc-linux-gnu*
 koreader-emulator-x86_64-apple-darwin*
 koreader-pocketbook-arm-obreey-linux-gnueabi*
 koreader-ubuntu-touch-arm-linux-gnueabihf*
+koreader-prst-arm-linux-gnueabihf*
```
Comment (NiLuJe) [inline .gitignore]: prst -> sony-prstux?

**Comment by Frenzie** [comment]
Nightlies are on GitLab, not here. :-)

Quickie on phone so no link. It should only be a couple mins to add.

PS Could you also add a few lines to kodev?

**Comment by NiLuJe** [comment]
I'm partial to a dash vs. the underscore, if only to match w/ ubuntu-touch & kindle-legacy ;).

(But that's nitpicking of the highest order ^^).

**Comment by v01d** [comment]
I would have to change that in base as well to match. Maybe leave it for later?

**Comment by NiLuJe** [comment]
I was speaking purely in terms of the filename of the package, as well as maaaaybe the name of the Make target, which I think is all here, not in base?

**Comment by v01d** [comment]
on base the TARGET variable is checked against. same for CI config

**Comment by NiLuJe** [comment]
Right, my bad, I missed that after the rename commit.

**Comment by v01d** [comment]
done with the rename

**Comment by Frenzie** [comment]
@v01d The moment of truth. I added it to GL and the first nightly is building [here](https://gitlab.com/koreader/nightly-builds/-/jobs/95847858) as we speak. It could easily take a half hour because there's no cache yet.

**Comment by v01d** [comment]
Seems to have worked =)

**Comment by v01d** [comment]
BTW, to have OTA available for this platform, would it simply work if I just added an entry in otamanager.lua? Or what do I need to do to support this?
The builds are the same as gitlab nightlies?

**Comment by Frenzie** [comment]
Yes, all that is done on the koreader.rocks server using this script (in a Docker container for easier portability):

https://github.com/koreader/koreader-misc/blob/3de3dc0a113ca374a853d9199c687c7993fc3767/nightswatcher/nightswatcher.py#L123-L125

You'll notice the Sony build was already matched by the regexp and put up on koreader.rocks: http://build.koreader.rocks/download/nightly/v2015.11-1842-g2eabf1d_2018-09-08/

But for OTA updates it'd need to be added to the OTA set as well.

Nothing difficult, just a bit annoying because you (or at least I) have to deal with 50/4, so uploading those Docker images takes a while.

**Comment by NiLuJe** [comment]
Ping me once that's done, so that I don't forget to update the cleanup script ;)

**Pull request merged**